docs/csharp/language-reference/operators/snippets/shared/NullForgivingOperator.cs

```diff
@@ -39,9 +39,7 @@ public static void Main()
         }
 
         public static bool IsValid(Person? person)
-        {
-            return person != null && !string.IsNullOrEmpty(person.Name);
-        }
+            => !string.IsNullOrEmpty(person?.Name);
```

Comment:
I'm curious about this change instead:

```csharp
public static bool IsValid(Person? person) => person is {​​​​​​​ Name: {​​​​​​​ Length: > 0 }​​​​​​​ }​​​​​​​;
```

Thoughts?

Reply:
It's a good example of pattern matching that in the context of this article may be a bit overwhelming (like I noticed above)

Reply:
What about `person is not null && person.Name is not null`
It's not refactoring, though, but would fit the article

Reply:
I like that version from @pkulikov It is semantically different, but does highlight the concepts for the article better than the existing code, or the other suggestions.

Reply:
> What about `person is not null && person.Name is not null`

Looks good. Actually, even `person is not null` would be enough for the article. But I went for your suggestion.